sonic-net/sonic-buildimage
# [Nvidia-bluefield] Add infrastructure for the DPU NIC FW auto upgrade

#### Why I did it
Extend Nvidia Bluefield SONiC infrastructure to support DPU NIC FW auto upgrade.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Extend the build system and init scripts to support the FW upgrade.

#### How to verify it
Compile an image with the new FW version. Run image installation. Verify that the running FW is upgraded after the image installation.



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
